### DISCUSSION TITLE
Divmod Nevow, Divmod Athena maybe?

### DISCUSSION THREAD

**Discussion by mpasternak** [volfpeter/htmy#24] [Ideas]
Long, long time ago in a galaxy far, far away, there was a technology long forgotten, called Divmod Nevow, https://divmod.readthedocs.io/en/latest/products/nevow/ . Your project reminds me of it. 

There was also Divmod Athena or the so-called widgets for Nevow. You declared a few things in the code, then wham, server could automagically receive RPC requests from the client. 

And both were async, because both based on Twisted Python. 

Take a look at Athena if you please, it was really cool way to make your app alive. https://github.com/MostAwesomeDude/divmod-docs/blob/master/products/nevow/athena/index.rst

**Comment by volfpeter** [comment]
Thanks for taking the time to share this, I'll have a look when I have time.
If I ever start working on features like client-server state sync or RPC, I'll most probably do it in another library that's built on top of this one (or maybe be rendering engine agnostic), just to keep this lib simple, flexible, and maintainable long-term. But this would be a complex task, it's hard to find enough spare time even to just correctly prototype it and create the basic interfaces without a few experienced contributors.
